parsers/IN_PB.py

```diff
@@ -84,6 +78,36 @@ def fetch_production(zone_key='IN-PB', session=None, target_datetime=None, logge
     return data
 
 
+def read_punjab_consumption_date(date_text, time_text, current):
+    """ Read Punjab consumption date, format of this is not consistent.
+        https://github.com/tmrowco/electricitymap/issues/854 """
+    date_time_string = date_text + time_text + 'Asia/Kolkata'
+
+    dates = []
+
+    try:
+        # Try to parse date string using MM/DD/YYYY format.
+        mm_dd_yyyy_format_date = get(date_time_string, "MM/DD/YYYYHH:mm:ssZZZ")
+        if mm_dd_yyyy_format_date and mm_dd_yyyy_format_date < current:
```

Comment:
We could avoid using the date at all.  Since this consumption parser can't get past dates we could just assume `india_current_date = utcnow().to('Asia/Kolkata')` is right and add the time to it. The problem comes if the page stops updating and we end up scraping the same data over and over again.

Reply:
As a safety check we could try and see if the parsed date is the same (or within 1 of the current).

Reply:
@corradio The proper date is selected after parse all cases.
```python
    if len(dates) > 0:
        return min(dates, key=lambda date: abs(date - current))
    else:
        raise Exception('IN-PB', 'Can''t read Punjab consumption date. DateTime String: {0}, Current Date: {1}'.format(date_time_string, current))
```

Reply:
@blackleg ah yes that is a good way of making sure no bad dates get accepted.